applications/googlehome-meets-dotnetcontainers/GoogleHomeAspNetCoreDemoServer/Dialogflow/Intents/Translation/TranslateEnglishHandler.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright(c) 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+using Google.Cloud.Dialogflow.V2;
+using Google.Cloud.Translation.V2;
+using System;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+
+namespace GoogleHomeAspNetCoreDemoServer.Dialogflow.Intents
+{
+    /// <summary>
+    /// Handler for "translation.english" DialogFlow intent.
+    /// </summary>
+    [Intent("translation.english")]
+    public class TranslateEnglishHandler : BaseHandler
+    {
+        /// <summary>
+        /// Initializes class with the conversation.
+        /// </summary>
+        /// <param name="conversation"></param>
+        public TranslateEnglishHandler(Conversation conversation) : base(conversation)
+        {
+        }
+
+        /// <summary>
+        /// Handle the intent.
+        /// </summary>
+        /// <param name="req">Webhook request</param>
+        /// <returns>Webhook response</returns>
+        public override async Task<WebhookResponse> HandleAsync(WebhookRequest req)
+        {
+            var errorMessage = ExtractAndValidateParameters(req, out string englishPhrase, out string languageCode);
+            if (errorMessage != null)
+            {
+                DialogflowApp.Show($"<div>{errorMessage}</div>");
+
+                return new WebhookResponse 
+                {
+                    FulfillmentText = errorMessage
+                };
+            }
+
+            var client = TranslationClient.Create();
+            var response = await client.TranslateTextAsync(englishPhrase, languageCode, LanguageCodes.English);
+
+            DialogflowApp.Show($"<div>'{englishPhrase}' is '{response.TranslatedText}' in {languageCode}</div>");
+            return new WebhookResponse { FulfillmentText = response.TranslatedText };
+        }
+
+        private string ExtractAndValidateParameters(WebhookRequest req, out string englishPhrase, out string languageCode)
```

Comment:
Please add a comment describing the return value.

Reply:
Added the comment